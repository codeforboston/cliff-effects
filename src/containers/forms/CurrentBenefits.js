import { connect } from 'react-redux';

import { CurrentBenefitsStep } from '../../forms/CurrentBenefits';
import { setHasBenefit } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),

    USState: state.getIn([
      'geography',
      'state',
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setHasBenefit({ benefit, value }) {
      dispatch(
        setHasBenefit({
          time: 'current',
          benefit,
          value,
        })
      );
    },
  };
};

const CurrentBenefitsContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentBenefitsStep);

export { CurrentBenefitsContainer as CurrentBenefitsStep };
