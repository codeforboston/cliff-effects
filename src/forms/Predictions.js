import React from 'react';
import { Form, Divider, Header, Tab, Message, Button, Menu } from 'semantic-ui-react';

// PROJECT COMPONENTS
import { FormPartsContainer } from './FormPartsContainer';
import { IntervalColumnHeadings } from '../components/headings';
import { CashFlowInputsRow } from './cashflow';
import { GraphHolder } from './output/GraphHolder';
import { BenefitsTable } from './output/BenefitsTable';
import { StackedBarGraph } from './output/StackedBarGraph';
import { StackedAreaGraph } from './output/StackedAreaGraph';
import { BenefitsLineGraph } from './output/BenefitsLineGraph';

// ========================================
// COMPONENTS
// ========================================
/** @todo description
*
* @function
* @param {object} props Values described below
* @property {object} props.future Client future/predictive data.
* @property {string} props.time Used in class names. Meant to make
*     this more easily decoupled in future.
* @property {function} props.updateClientValue Update client state
*     values.
*
* @returns {class} Component
*/
const IncomeForm = function ({ future, time, updateClientValue, snippets }) {

  var type = 'income';

  /**
  * As per Project Hope input, for the first prototype we're only
  * including the ability to change earned income.
  */
  return (
    <div className='field-aligner two-column'>
      <IntervalColumnHeadings type={ type } />
      <CashFlowInputsRow
        timeState={ future }
        type={ type }
        time={ time }
        updateClientValue = { updateClientValue }
        generic='earned'
        labelInfo='(Weekly income = hourly wage times average number of work hours per week)'>
        { snippets.i_futureIncomeQuestion }
      </CashFlowInputsRow>
    </div>
  );
};  // End IncomeForm() Component


const TabbedVisualizations = ({ client, snippets }) => {
  return (
  // Benefit Courses, Tracks, Routes, Traces, Progressions, Progress, Trajectories, Changes
    <Tab
      menu={{ color: 'teal',  attached: true, tabular: true }}
      panes={ [
        { 
          menuItem: (
            <Menu.Item key="tab1">
              { snippets.i_tabTitleChanges }
            </Menu.Item>
          ),
          render: () => {
            return (
              <Tab.Pane>
                <BenefitsTable
                  client={ client }
                  snippets={ snippets } />
              </Tab.Pane>
            );
          },
        },
        { 
          menuItem: (
            <Menu.Item key="tab2">
              { snippets.i_tabTitleChangesChart }
            </Menu.Item>
          ),  
          render: () => {return <Tab.Pane><StackedBarGraph client={ client } /></Tab.Pane>;}, 
        },
        {
          menuItem: (
            <Menu.Item key="tab3">
              { snippets.i_tabTitleStackedIncomes }
            </Menu.Item>
          ),
          render: () => {
            return (
              <Tab.Pane>
                <GraphHolder
                  client={ client }
                  Graph={ StackedAreaGraph } />
              </Tab.Pane>
            );
          },
        },
        {
          menuItem: (
            <Menu.Item key="tab4">
              { snippets.i_tabTitleBenefitPrograms }
            </Menu.Item>
          ),
          render: () => {
            return (
              <Tab.Pane>
                <GraphHolder
                  client={ client }
                  Graph={ BenefitsLineGraph } />
              </Tab.Pane>
            );
          },
        },
      ] } />
  );
};

/** @todo Abstract all the step components?
 *
 * @function
 * @param {object} props See below.
 * @property {function} props.updateClientValue Updates state upstream.
 * @property {function} props.translate Uses user chosen language-specific
 *    snippets.
 * @property {object} props.client JSON object with future and current values.
 * @property {function} props.nextStep Go to next form section.
 * @property {function} props.previousStep Go to previous form section.
 *
 * @returns {object} Component
 */
const PredictionsStep = function ({ updateClientValue, navData, client, snippets, openFeedback }) {

  /** @todo Are these titles accurate now? */
  return (
    <Form className = 'income-form flex-item flex-column'>
      <FormPartsContainer
        title     = { snippets.i_title }
        clarifier = { null }
        navData   = { navData }>
        <IncomeForm
          updateClientValue = { updateClientValue }
          future            = { client.future }
          time              = { 'future' } 
          snippets          = { snippets } />
        <Divider className='ui section divider hidden' />
        <Header
          as        ='h3'
          className ='ui Header align centered'>
          { snippets.i_chartsHeader }
        </Header>
        <Message
          visible
          warning
          style={{ 'textAlign': 'center' }}>
          { snippets.i_warningMessage }
          <br />
          <Button
            fluid
            color='teal'
            style={{
              'display':     'block',
              'marginLeft':  'auto',
              'marginRight': 'auto',
              'marginTop':   '10px',
              'maxWidth':    '400px', 
            }}
            onClick={ openFeedback }>
            { snippets.i_submitFeedback }
          </Button>
        </Message>
        <TabbedVisualizations 
          client   = { client }
          snippets = { snippets } />
      </FormPartsContainer>
    </Form>
  );
};  // End FutureIncomeStep() Component

export { PredictionsStep };
                                              
