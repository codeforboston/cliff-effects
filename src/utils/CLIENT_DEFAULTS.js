import { cloneDeep } from 'lodash';

/**
 * Default client values.
 * MUST ALWAYS BE VALID
 * 
 * Note: A 'positive number' money value can have more than
 *     two decimal places.
 */
const CLIENT_DEFAULTS = {

  // Property Name                | Default Value   | Valid Value Description

  current: {
    // Current programs
    hasSnap:                        false,            // boolean
    hasHousing:                     false,            // boolean
    // Household
    // 'm_' for 'member'
    household:  [{                  // 'head' member  // one or more member objects
        m_age:                      30,               // whole number
        m_role:                     'head',           // 'head', 'spouse', 'member'
        m_disabled:                 false             // boolean
    }],
    // Income
    earned:                         0,                // positive number (can have more than two decimal places)
    TAFDC:                          0,                // positive number
    SSI:                            0,                // positive number
    SSDI:                           0,                // positive number
    childSupportIn:                 0,                // positive number
    unemployment:                   0,                // positive number
    workersComp:                    0,                // positive number
    pension:                        0,                // positive number
    socialSecurity:                 0,                // positive number
    alimony:                        0,                // positive number
    otherIncome:                    0,                // positive number
    incomeExclusions:               0,                // positive number
    // Expenses
    childDirectCare:                0,                // positive number
    childBeforeAndAfterSchoolCare:  0,                // positive number
    childTransportation:            0,                // positive number
    childOtherCare:                 0,                // positive number
    earnedBecauseOfChildCare:       0,                // positive number
    childSupportPaidOut:            0,                // positive number
    adultDirectCare:                0,                // positive number
    adultTransportation:            0,                // positive number
    adultOtherCare:                 0,                // positive number
    disabledAssistance:             0,                // positive number
    earnedBecauseOfAdultCare:       0,                // positive number
    disabledMedical:                0,                // positive number
    otherMedical:                   0,                // positive number
    /** @todo When client has section 8, switch this to 'housingVoucher' */
    /** @todo Make shelter expense values more robust. */
    shelter:                        'homeless',       // 'homeless', 'homeowner', 'renter', 'housingVoucher'
    contractRent:                   0,                // positive number
    rentShare:                      0,                // positive number
    rent:                           0,                // positive number
    mortgage:                       0,                // positive number
    housingInsurance:               0,                // positive number
    propertyTax:                    0,                // positive number
    climateControl:                 false,            // boolean
    nonHeatElectricity:             false,            // boolean
    phone:                          false,            // boolean
    fuelAssistance:                 false,            // boolean  Note: Input is 'yes', 'no', but should be being converted
    otherExpenses:                  0                 // positive number
  },

  // Note: A `future` value will default to the client's `current` value until a user changes it explicitly
  future: {},                       // same structure // same

};

CLIENT_DEFAULTS.future = cloneDeep( CLIENT_DEFAULTS.current );


export { CLIENT_DEFAULTS };
