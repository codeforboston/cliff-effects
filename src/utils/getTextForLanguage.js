/** Returns a translator based on the language given */

// TRANSFORMER FUNCTIONS
import { mergeWith } from 'lodash';
import { interpolateSnippets } from './interpolation.js';

// DATA
import { getLocalizationData } from '../localization/all';
import inlineComponents from '../localization/inlineComponents';

// Get copy of localization data
const localizations = getLocalizationData();

// store interpolated and (if necessary) merged snippets objects
let finishedSnippets = { en: interpolateSnippets(localizations.en, inlineComponents) };

/** Customizes Lodash's mergeWith function to replace arrays completely
 * (to avoid arrays of English strings being mixed with arrays of translated
 * strings, if they happen to have different lengths).
 */
const mergeCustomizer = function (objValue, srcValue) {
  if (Array.isArray(objValue)) {
    return srcValue;
  }
};

/** Returns the object named by langName that contains
 * the text snippets of that language. If that language
 * doesn't exist, it warns the coder and returns English.
 */
const getTextForLanguage = function (langCode) {
  if (!localizations[ langCode ]) {
    console.warn('There\'s no localization for ' + langCode + '. Defaulting to English.');
    langCode = 'en';
  }

  if (!finishedSnippets[ langCode ]) {
    // interpolate snippets and merge with English (filling in any gaps)
    const interpolatedSnippets = interpolateSnippets(localizations[ langCode ], inlineComponents);

    // deeply merge the object containing snippets in langName with English,
    // so that we fall back to English if a particular field is missing.
    finishedSnippets[ langCode ] =  mergeWith({}, finishedSnippets.en, interpolatedSnippets, mergeCustomizer);
  }

  return finishedSnippets[ langCode ];
};  // End getTextForLanguage()


export { getTextForLanguage };
