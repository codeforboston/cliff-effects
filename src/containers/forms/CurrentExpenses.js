import { connect } from 'react-redux';

import { CurrentExpensesStep } from '../../forms/CurrentExpenses';

import { setCashValue, setHousingType, setPaysUtility, setGetsFuelAssistance } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setExpenseValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'current',
          name,
          value,
        })
      );
    },

    setHousingType({ housingType }) {
      dispatch(
        setHousingType({
          time: 'current',
          housingType,
        })
      );
    },

    setPaysUtility({ utility, paysUtility }) {
      dispatch(
        setPaysUtility({
          time: 'current',
          utility,
          paysUtility,
        })
      );
    },

    setGetsFuelAssistance({ getsAssistance }) {
      dispatch(
        setGetsFuelAssistance({
          time: 'current',
          getsAssistance,
        })
      );
    },
  };
};

const CurrentExpensesStepContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentExpensesStep);

export { CurrentExpensesStepContainer as CurrentExpensesStep };
