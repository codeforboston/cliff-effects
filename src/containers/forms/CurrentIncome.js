import { connect } from 'react-redux';

import { CurrentIncomeStep } from '../../forms/CurrentIncome';

import { setCashValue } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setIncomeValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'current',
          name,
          value,
        })
      );
    },
  };
};

const CurrentIncomeStepContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentIncomeStep);

export { CurrentIncomeStepContainer as CurrentIncomeStep };
