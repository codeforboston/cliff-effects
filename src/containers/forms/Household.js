import { connect } from 'react-redux';

import { HouseholdStep } from '../../forms/Household';
import {
  addMember,
  removeMember,
  setMemberIsDisabled,
  setMemberRole,
  setMemberAge,
} from '../../actions';

const time = 'current';

const mapStateToProps = (state) => {
  return {
    household: state.getIn([
      'client',
      'current',
      'household', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setMemberAge({ index, age }) {
      dispatch(
        setMemberAge({
          time,
          index,
          age,
        })
      );
    },

    setMemberIsDisabled({ index, isDisabled }) {
      dispatch(
        setMemberIsDisabled({
          time,
          index,
          isDisabled,
        })
      );
    },

    setMemberRole({ index, role }) {
      dispatch(
        setMemberRole({
          time,
          index,
          role,
        })
      );
    },

    removeMember({ index }) {
      dispatch(
        removeMember({
          time,
          index,
        })
      );
    },

    addMember({ member }) {
      dispatch(
        addMember({
          time,
          member,
        })
      );
    },
  };
};

const HouseholdStepContainer = connect(mapStateToProps, mapDispatchToProps)(HouseholdStep);

export { HouseholdStepContainer as HouseholdStep };
