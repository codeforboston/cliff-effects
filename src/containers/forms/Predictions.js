import { connect } from 'react-redux';

import { PredictionsStep } from '../../forms/Predictions';

import { setCashValue } from '../../actions';

const mapStateToProps = (state) => {
  return { client: state.get('client') };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setPredictionValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'future',
          name,
          value,
        })
      );
    },
  };
};

const PredictionsStepContainer = connect(mapStateToProps, mapDispatchToProps)(PredictionsStep);

export { PredictionsStepContainer as PredictionsStep };
