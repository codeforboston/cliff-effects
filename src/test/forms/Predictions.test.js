import React from 'react';
import { shallow } from 'enzyme';

import { PredictionsStep } from '../../forms/Predictions';

import { CLIENT_DEFAULTS } from '../../utils/CLIENT_DEFAULTS';

// LOCALIZATION
import { getTextForLanguage } from '../../utils/getTextForLanguage';


var snippets = getTextForLanguage(`en`);
test('Prediction component renders as snapshot correctly', () => {
  const navData = {
          left:   { text: 'Previous', onClick: jest.fn() },
          middle: null,
          right:  { text: 'New Client', onClick: jest.fn() },
        },
        updateClientValue     = jest.fn(),
        saveForm         = jest.fn(),
        askToResetClient = jest.fn(),
        openFeedback     = jest.fn(),
        formSnippets     = snippets.visitPage.predictions;

  const wrapper = shallow(
    <PredictionsStep
      currentStep       = { 5 }
      client            = { CLIENT_DEFAULTS }
      navData           = { navData }
      updateClientValue      = { updateClientValue }
      saveForm          = { saveForm }
      askToResetClient  = { askToResetClient }
      openFeedback      = { openFeedback }
      snippets          = { formSnippets } />
  );

  expect(wrapper).toMatchSnapshot();
});
