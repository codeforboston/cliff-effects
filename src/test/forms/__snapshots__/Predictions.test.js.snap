// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prediction component renders as snapshot correctly 1`] = `
<Form
  as="form"
  className="income-form flex-item flex-column"
>
  <FormPartsContainer
    clarifier={null}
    navData={
      Object {
        "left": Object {
          "onClick": [Function],
          "text": "Previous",
        },
        "middle": null,
        "right": Object {
          "onClick": [Function],
          "text": "New Client",
        },
      }
    }
    title={
      <span
        className="snippet"
        lang="en"
      >
        What Might Happen?
      </span>
    }
  >
    <IncomeForm
      future={
        Object {
          "SSDI": 0,
          "SSI": 0,
          "TAFDC": 0,
          "adultDirectCare": 0,
          "adultOtherCare": 0,
          "adultTransportation": 0,
          "alimony": 0,
          "childBeforeAndAfterSchoolCare": 0,
          "childDirectCare": 0,
          "childOtherCare": 0,
          "childSupportIn": 0,
          "childSupportPaidOut": 0,
          "childTransportation": 0,
          "climateControl": false,
          "contractRent": 0,
          "disabledAssistance": 0,
          "disabledMedical": 0,
          "earned": 0,
          "earnedBecauseOfAdultCare": 0,
          "earnedBecauseOfChildCare": 0,
          "fuelAssistance": false,
          "hasSection8": false,
          "hasSnap": false,
          "household": Array [
            Object {
              "m_age": 30,
              "m_disabled": false,
              "m_role": "head",
            },
          ],
          "housing": "homeless",
          "housingInsurance": 0,
          "mortgage": 0,
          "nonHeatElectricity": false,
          "otherExpensesCable": 0,
          "otherExpensesCareProducts": 0,
          "otherExpensesClothes": 0,
          "otherExpensesEntertainment": 0,
          "otherExpensesFood": 0,
          "otherExpensesMedical": 0,
          "otherExpensesOther": 0,
          "otherExpensesPhone": 0,
          "otherExpensesTransport": 0,
          "otherExpensesUtilities": 0,
          "otherIncome": 0,
          "otherMedical": 0,
          "pension": 0,
          "phone": false,
          "propertyTax": 0,
          "rent": 0,
          "rentShare": 0,
          "socialSecurity": 0,
          "unemployment": 0,
          "wantsToSeeOtherExpenses": false,
          "workersComp": 0,
        }
      }
      snippets={
        Object {
          "i_benefitsTableTitle": <span
            className="snippet"
            lang="en"
        >
            Changes
        </span>,
          "i_chartsHeader": <span
            className="snippet"
            lang="en"
        >
            With the new pay, how could your benefits change?
        </span>,
          "i_columnBenefit": <span
            className="snippet"
            lang="en"
        >
            Benefit
        </span>,
          "i_columnCurrentBenefits": <span
            className="snippet"
            lang="en"
        >
            Current Benefits
        </span>,
          "i_columnDifference": <span
            className="snippet"
            lang="en"
        >
            Difference
        </span>,
          "i_columnNewEstimate": <span
            className="snippet"
            lang="en"
        >
            New Estimate
        </span>,
          "i_dollarSign": <span
            className="snippet"
            lang="en"
        >
            $
        </span>,
          "i_futureIncomeQuestion": <span
            className="snippet"
            lang="en"
        >
            How much money would you get paid in the future? (You can try different amounts)
        </span>,
          "i_perMonth": <span
            className="snippet"
            lang="en"
        >
             / month
        </span>,
          "i_rowIncome": <span
            className="snippet"
            lang="en"
        >
            Income
        </span>,
          "i_rowNetTotal": <span
            className="snippet"
            lang="en"
        >
            Net Total
        </span>,
          "i_rowSNAP": <span
            className="snippet"
            lang="en"
        >
            SNAP
        </span>,
          "i_rowSection8": <span
            className="snippet"
            lang="en"
        >
            Section 8 Housing
        </span>,
          "i_rowTotalBenefits": <span
            className="snippet"
            lang="en"
        >
            Total Benefits
        </span>,
          "i_submitFeedback": <span
            className="snippet"
            lang="en"
        >
            Submit Feedback
        </span>,
          "i_tabTitleBenefitPrograms": <span
            className="snippet"
            lang="en"
        >
            Benefit Programs
        </span>,
          "i_tabTitleChanges": <span
            className="snippet"
            lang="en"
        >
            Changes
        </span>,
          "i_tabTitleChangesChart": <span
            className="snippet"
            lang="en"
        >
            Changes Chart
        </span>,
          "i_tabTitleStackedIncomes": <span
            className="snippet"
            lang="en"
        >
            Stacked Incomes
        </span>,
          "i_title": <span
            className="snippet"
            lang="en"
        >
            What Might Happen?
        </span>,
          "i_warningMessage": <span
            className="snippet"
            lang="en"
        >
            This tool is in testing and these numbers might not be right. If they're not, we'd appreciate your feedback.
        </span>,
          "langCode": "en",
        }
      }
      time="future"
      updateClientValue={[Function]}
    />
    <Divider
      className="ui section divider hidden"
    />
    <Header
      as="h3"
      className="ui Header align centered"
    >
      <span
        className="snippet"
        key="en229"
        lang="en"
      >
        With the new pay, how could your benefits change?
      </span>
    </Header>
    <Message
      style={
        Object {
          "textAlign": "center",
        }
      }
      visible={true}
      warning={true}
    >
      <span
        className="snippet"
        key="en230"
        lang="en"
      >
        This tool is in testing and these numbers might not be right. If they're not, we'd appreciate your feedback.
      </span>
      <br />
      <Button
        as="button"
        color="teal"
        fluid={true}
        onClick={[Function]}
        style={
          Object {
            "display": "block",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "10px",
            "maxWidth": "400px",
          }
        }
      >
        <span
          className="snippet"
          key="en231"
          lang="en"
        >
          Submit Feedback
        </span>
      </Button>
    </Message>
    <TabbedVisualizations
      client={
        Object {
          "current": Object {
            "SSDI": 0,
            "SSI": 0,
            "TAFDC": 0,
            "adultDirectCare": 0,
            "adultOtherCare": 0,
            "adultTransportation": 0,
            "alimony": 0,
            "childBeforeAndAfterSchoolCare": 0,
            "childDirectCare": 0,
            "childOtherCare": 0,
            "childSupportIn": 0,
            "childSupportPaidOut": 0,
            "childTransportation": 0,
            "climateControl": false,
            "contractRent": 0,
            "disabledAssistance": 0,
            "disabledMedical": 0,
            "earned": 0,
            "earnedBecauseOfAdultCare": 0,
            "earnedBecauseOfChildCare": 0,
            "fuelAssistance": false,
            "hasSection8": false,
            "hasSnap": false,
            "household": Array [
              Object {
                "m_age": 30,
                "m_disabled": false,
                "m_role": "head",
              },
            ],
            "housing": "homeless",
            "housingInsurance": 0,
            "mortgage": 0,
            "nonHeatElectricity": false,
            "otherExpensesCable": 0,
            "otherExpensesCareProducts": 0,
            "otherExpensesClothes": 0,
            "otherExpensesEntertainment": 0,
            "otherExpensesFood": 0,
            "otherExpensesMedical": 0,
            "otherExpensesOther": 0,
            "otherExpensesPhone": 0,
            "otherExpensesTransport": 0,
            "otherExpensesUtilities": 0,
            "otherIncome": 0,
            "otherMedical": 0,
            "pension": 0,
            "phone": false,
            "propertyTax": 0,
            "rent": 0,
            "rentShare": 0,
            "socialSecurity": 0,
            "unemployment": 0,
            "wantsToSeeOtherExpenses": false,
            "workersComp": 0,
          },
          "future": Object {
            "SSDI": 0,
            "SSI": 0,
            "TAFDC": 0,
            "adultDirectCare": 0,
            "adultOtherCare": 0,
            "adultTransportation": 0,
            "alimony": 0,
            "childBeforeAndAfterSchoolCare": 0,
            "childDirectCare": 0,
            "childOtherCare": 0,
            "childSupportIn": 0,
            "childSupportPaidOut": 0,
            "childTransportation": 0,
            "climateControl": false,
            "contractRent": 0,
            "disabledAssistance": 0,
            "disabledMedical": 0,
            "earned": 0,
            "earnedBecauseOfAdultCare": 0,
            "earnedBecauseOfChildCare": 0,
            "fuelAssistance": false,
            "hasSection8": false,
            "hasSnap": false,
            "household": Array [
              Object {
                "m_age": 30,
                "m_disabled": false,
                "m_role": "head",
              },
            ],
            "housing": "homeless",
            "housingInsurance": 0,
            "mortgage": 0,
            "nonHeatElectricity": false,
            "otherExpensesCable": 0,
            "otherExpensesCareProducts": 0,
            "otherExpensesClothes": 0,
            "otherExpensesEntertainment": 0,
            "otherExpensesFood": 0,
            "otherExpensesMedical": 0,
            "otherExpensesOther": 0,
            "otherExpensesPhone": 0,
            "otherExpensesTransport": 0,
            "otherExpensesUtilities": 0,
            "otherIncome": 0,
            "otherMedical": 0,
            "pension": 0,
            "phone": false,
            "propertyTax": 0,
            "rent": 0,
            "rentShare": 0,
            "socialSecurity": 0,
            "unemployment": 0,
            "wantsToSeeOtherExpenses": false,
            "workersComp": 0,
          },
        }
      }
      snippets={
        Object {
          "i_benefitsTableTitle": <span
            className="snippet"
            lang="en"
        >
            Changes
        </span>,
          "i_chartsHeader": <span
            className="snippet"
            lang="en"
        >
            With the new pay, how could your benefits change?
        </span>,
          "i_columnBenefit": <span
            className="snippet"
            lang="en"
        >
            Benefit
        </span>,
          "i_columnCurrentBenefits": <span
            className="snippet"
            lang="en"
        >
            Current Benefits
        </span>,
          "i_columnDifference": <span
            className="snippet"
            lang="en"
        >
            Difference
        </span>,
          "i_columnNewEstimate": <span
            className="snippet"
            lang="en"
        >
            New Estimate
        </span>,
          "i_dollarSign": <span
            className="snippet"
            lang="en"
        >
            $
        </span>,
          "i_futureIncomeQuestion": <span
            className="snippet"
            lang="en"
        >
            How much money would you get paid in the future? (You can try different amounts)
        </span>,
          "i_perMonth": <span
            className="snippet"
            lang="en"
        >
             / month
        </span>,
          "i_rowIncome": <span
            className="snippet"
            lang="en"
        >
            Income
        </span>,
          "i_rowNetTotal": <span
            className="snippet"
            lang="en"
        >
            Net Total
        </span>,
          "i_rowSNAP": <span
            className="snippet"
            lang="en"
        >
            SNAP
        </span>,
          "i_rowSection8": <span
            className="snippet"
            lang="en"
        >
            Section 8 Housing
        </span>,
          "i_rowTotalBenefits": <span
            className="snippet"
            lang="en"
        >
            Total Benefits
        </span>,
          "i_submitFeedback": <span
            className="snippet"
            lang="en"
        >
            Submit Feedback
        </span>,
          "i_tabTitleBenefitPrograms": <span
            className="snippet"
            lang="en"
        >
            Benefit Programs
        </span>,
          "i_tabTitleChanges": <span
            className="snippet"
            lang="en"
        >
            Changes
        </span>,
          "i_tabTitleChangesChart": <span
            className="snippet"
            lang="en"
        >
            Changes Chart
        </span>,
          "i_tabTitleStackedIncomes": <span
            className="snippet"
            lang="en"
        >
            Stacked Incomes
        </span>,
          "i_title": <span
            className="snippet"
            lang="en"
        >
            What Might Happen?
        </span>,
          "i_warningMessage": <span
            className="snippet"
            lang="en"
        >
            This tool is in testing and these numbers might not be right. If they're not, we'd appreciate your feedback.
        </span>,
          "langCode": "en",
        }
      }
    />
  </FormPartsContainer>
</Form>
`;
